ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -192,12 +198,208 @@ static INT_PTR CALLBACK ShowMessageExtProc(HWND hwndDlg, UINT uMsg, WPARAM wPara
   return 0;
 }
 
-static INT CALLBACK GetDirectoryAltProc(HWND hwnd, UINT uMsg, LPARAM lp, LPARAM pData)
+WCHAR wstrPromptStr[4096];
+WCHAR wstrTextEntry[MAX_PATH];
+WCHAR wstrWindowStr[MAX_PATH];
+bool HideInput = 0;
+
+void ClientResize(HWND hWnd, int nWidth, int nHeight)
 {
-  if (uMsg == BFFM_INITIALIZED)
-    SetWindowText(hwnd, gs_cap.c_str());
+  RECT rcClient, rcWind;
+  POINT ptDiff;
+  GetClientRect(hWnd, &rcClient);
+  GetWindowRect(hWnd, &rcWind);
+  ptDiff.x = (rcWind.right - rcWind.left) - rcClient.right;
+  ptDiff.y = (rcWind.bottom - rcWind.top) - rcClient.bottom;
+  MoveWindow(hWnd, rcWind.left, rcWind.top, nWidth + ptDiff.x, nHeight + ptDiff.y, TRUE);
+}
 
-  return 0;
+LRESULT CALLBACK InputBoxHookProc(HWND hWndDlg, UINT Msg, WPARAM wParam, LPARAM lParam)
+{
+  switch (Msg)
+  {
+  case WM_INITDIALOG:
+    ClientResize(hWndDlg, 357, 128);
+    RECT rect;
+    GetWindowRect(hWndDlg, &rect);
+    MoveWindow(hWndDlg,
+      (GetSystemMetrics(SM_CXSCREEN) / 2) - ((rect.right - rect.left) / 2),
+      (GetSystemMetrics(SM_CYSCREEN) / 3) - ((rect.bottom - rect.top) / 3),
+      rect.right - rect.left, rect.bottom - rect.top, TRUE);
+    MoveWindow(GetDlgItem(hWndDlg, IDOK), 272, 10, 75, 23, TRUE);
+    MoveWindow(GetDlgItem(hWndDlg, IDCANCEL), 272, 39, 75, 23, TRUE);
+    MoveWindow(GetDlgItem(hWndDlg, 990), 11, 94, 336, 23, TRUE);
+    MoveWindow(GetDlgItem(hWndDlg, 992), 11, 11, 252, 66, TRUE);
+    SetWindowLongPtr(hWndDlg, GWL_EXSTYLE, GetWindowLongPtr(hWndDlg, GWL_EXSTYLE) | WS_EX_DLGMODALFRAME);
+    SetWindowPos(hWndDlg, 0, 0, 0, 0, 0, SWP_FRAMECHANGED | SWP_NOMOVE | SWP_NOSIZE | SWP_NOZORDER);
+    SetDlgItemTextW(hWndDlg, 992, wstrPromptStr);
+    SetDlgItemTextW(hWndDlg, 990, wstrTextEntry);
+    SetWindowTextW(hWndDlg, wstrWindowStr);
+    if (HideInput == 1)
+      SendDlgItemMessage(hWndDlg, 990, EM_SETPASSWORDCHAR, '*', NULL);
+    SendDlgItemMessage(hWndDlg, 990, EM_SETSEL, '*', NULL);
+    SendDlgItemMessage(hWndDlg, 990, WM_SETFOCUS, NULL, NULL);
+    return TRUE;
+
+  case WM_COMMAND:
+    switch (wParam)
+    {
+    case IDOK:
+      GetDlgItemTextW(hWndDlg, 990, wstrTextEntry, MAX_PATH);
+      gs_form_canceled = 0;
+      EndDialog(hWndDlg, 0);
+      return TRUE;
+    case IDCANCEL:
+      tstring wstrEmpty = widen("");
+      wcsncpy(wstrTextEntry, wstrEmpty.c_str(), MAX_PATH);
+      gs_form_canceled = 1;
+      EndDialog(hWndDlg, 0);
+      return TRUE;
+    }
+    break;
+  }
+
+  return FALSE;
+}
+
+WCHAR *LowerCaseToActualPathName(WCHAR *wstr_dname)
+{
+  LPITEMIDLIST pstr_dname;
+  SHParseDisplayName(wstr_dname, 0, &pstr_dname, 0, 0);
+
+  static WCHAR wstr_result[MAX_PATH];
+  SHGetPathFromIDListW(pstr_dname, wstr_result);
+  return wstr_result;
+}
+
+WCHAR wstr_dname[MAX_PATH];
+WCHAR wstr_label5[MAX_PATH];
+tstring ActualPath;
+HWND label6;
+
+UINT APIENTRY OFNHookProcOldStyle(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+  if (uMsg == WM_INITDIALOG)
```

Comment:
It's standard practice (and just generally a good idea for readability reasons) to use a `switch` statement for uMsg in a `WndProc`/`HookProc`. Actually, you're doing it above... why not here?

Reply:
I've tried making this a switch statement and the dialog doesn't even show up. I have it this way for a reason. It is wayy more confusing with a switch statement imo. I didn't know switch statments could ever be less confusing that an if-then.

Reply:
You need to have a break after each case. If you omit that, it'll do multiple (incorrect) things for any given case except the last.